apps/src/JSInterpreter.js

```diff
@@ -423,8 +433,10 @@ JSInterpreter.prototype.executeInterpreter = function (firstStep, runUntilCallba
     }
     var err = safeStepInterpreter(this);
     if (!err) {
-      doneUserLine = doneUserLine ||
-        (inUserCode && this.interpreter.stateStack[0] && this.interpreter.stateStack[0].done);
+      var nodeType = this.interpreter.stateStack[0].node.type;
+      if (inUserCode) {
+        doneUserLine = doneUserLine || INTERSTITIAL_NODES.hasOwnProperty(nodeType);
```

Comment:
`hasOwnProperty` won't go up the prototype chain on misses, but I don't think we expect to have any extra junk on `Object.prototype`.  I'll change.